osgbuild/utils.py

```diff
@@ -390,3 +390,32 @@ def print_table(columns_by_header):
 def is_url(location):
     return re.match(r'[-a-z+]+://', location)
 
+
+# Functions for manipulating a directory stack in the style of bash
+# pushd/popd.
+__dir_stack = []
+
+def pushd(new_dir):
+    """Change the current working directory to `new_dir`, and push the
+    old one onto the directory stack `__dir_stack`.
+    """
+    global __dir_stack
```

Comment:
Note that `global` shouldn't be needed for `__dir_stack`, since you never assign to it (with `=`).